testtools/UART_interface/esp_arduino_uart_interface.py

```diff
@@ -0,0 +1,159 @@
+import os
+import sys
+import time
+try:
+    from testtools.UART_interface.base_uart_interface import uart_interface
+except:
+    from base_uart_interface import uart_interface
+
+try:
+    import testtools.UART_interface.azure_test_firmware_errors as azure_test_firmware_errors
+    import testtools.UART_interface.serial_settings as serial_settings
+    import testtools.UART_interface.serial_commands_dict as commands_dict
+except:
+    import azure_test_firmware_errors
+    import serial_settings
+    import serial_commands_dict as commands_dict
+
+
+# ------- global usecase fcns -------
+def device_setup():
+    pass
+
+def check_sdk_errors(line):
+    if "rror" in line or "ailed" in line or "ailure" in line:
```

Comment:
maybe make use of `.casefold` here?
[Link](https://www.tutorialspoint.com/casefold-string-in-python)
maybe:
```python
line_to_lower = line.casefold()
if "error" in line_to_lower or "fail" in line_to_lower
```

Reply:
I'm going to start with lower(), and see if it works. I don't want to do too much processing per line.